src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/TelemetryReporterTests.cs

```diff
@@ -322,4 +322,40 @@ public void TrackLspRequest()
                 Assert.Equal(correlationId, correlationProperty.Value);
             });
     }
+
+    [Fact]
+    public void ReportFault_InnerMostExceptionIsOperationCanceledException_SkipsFaultReport()
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var innerMostException = new OperationCanceledException();
+        var exception = CreateNestedException(typeof(OperationCanceledException), 3, innerMostException);
+
+        // Act
+        reporter.ReportFault(exception, "Test message");
+
+        // Assert
+        Assert.Empty(reporter.Events);
+    }
+
+    private Exception CreateNestedException(Type exceptionType, int depth, Exception? innerMostException)
+    {
+        var exception = CreateException(exceptionType, "Test", innerMostException);
+        for (var i = 0; i < depth; i++)
+        {
+            exception = CreateException(exceptionType, "Test", exception);
+        }
+
+        return exception;
+    }
+
+    private Exception CreateException(Type exceptionType, string message, Exception? innerException)
+    {
+        if (exceptionType == typeof(OperationCanceledException))
+        {
+            return new OperationCanceledException(message, innerException);
+        }
+
+        throw new NotImplementedException();
```

Comment:
I am completing the implementation of this now based on what will be needed for https://github.com/dotnet/razor/pull/9919